### ISSUE TITLE
[Feature] Detailed test execution log with timing information

### ISSUE THREAD

**Issue by dgozman** [microsoft/playwright#15466]
If someone is interested in all the details about the test run, we can provide some kind of a detailed log. Some examples of what can be included:
- Which fixtures were instantiated for what test and in what order?
- Which hooks were run for a particular test?
- How long did the worker setup and teardown take? Global setup and teardown? Fixtures?
- What tests a particular worker process did run?

**Comment by vsravuri** [comment]
@dgozman It will be helpful if the timestamp of test case execution (start time) is also included, this will help to analyze application log files for failed test cases. My e2e test runs for 4 to 5 hours, i find it difficult to match the failed test case with application log files.

**Comment by paul-sachs** [comment]
I've recently noticed my tests going from 2s to 2mins and thought this would be helpful to find out why. Weirdly still can't figure it out what was eating up all that extra time.

**Comment by Oscaruzzo** [comment]
I agree with @vsravuri that timestamps are essential when I need to check my application logs. I'm using a custom reporter at the moment.

**Comment by amit-watershed** [comment]
Would like to plus one on this feature request as well

**Comment by cullylarson** [comment]
Not as details as the OP is describing, but if you need slightly more detail: https://playwright.dev/docs/test-reporters

**Comment by Palash9088** [comment]
> I've recently noticed my tests going from 2s to 2mins and thought this would be helpful to find out why. Weirdly still can't figure it out but it would be useful to see what was eating up all that extra time.

Hi was this resolved ?

**Comment by vitalets** [comment]
I've built [playwright-timeline-reporter](https://github.com/vitalets/playwright-timeline-reporter), which captures detailed timing information for tests, hooks, fixtures, and workers, and visualizes it as an interactive timeline:

<a href="https://vitalets.github.io/playwright-timeline-reporter/demos/index.html">
<img alt="timeline reporter" src="https://github.com/user-attachments/assets/926909f2-9b89-4e3b-805d-28ff3e1cb2d6" />
</a>

Feel free to try it and share your feedback.
